routers/api/v1/repo/repo.go

```diff
@@ -15,9 +16,43 @@ import (
 	"code.gitea.io/gitea/modules/context"
 	"code.gitea.io/gitea/modules/log"
 	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/util"
 	"code.gitea.io/gitea/routers/api/v1/convert"
 )
 
+// SearchRepoOption options when searching repositories
+// swagger:parameters repoSearch
+type SearchRepoOption struct { // TODO: Move SearchRepoOption to Gitea SDK
+	// Keyword to search
+	//
+	// in: query
+	Keyword string `json:"q"`
+	// Repository owner to search
+	//
+	// in: query
+	OwnerID int64 `json:"uid"`
+	// Limit of result
+	//
+	// maximum: setting.ExplorePagingNum
+	// in: query
+	PageSize int `json:"limit"`
+	// Type of repository to search, related to owner
+	//
+	// in: query
+	SearchMode string `json:"mode"`
+}
+
+// searchMode is repository filtering mode identifier
+type searchMode int
+
+const (
```

Comment:
I don't mean to be difficult, but adding a enum just for this one function seems like overkill. From what I can tell, the enum is used in three places:
1. In `validateSearchInput()` to check that the user provided a valid mode
2. In `validateSearchInput()` to check that the user provided a uid for certain modes
3. In assigning `opts.Collaborate`

Could we do the following?
- Add a `default` case to the `switch modeQuery { ...` to handle the case of a user providing an invalid mode
- Add check to the `case "source"` and `case "mirror"` cases to handle the case of a missing uid
  - It is unclear to me why such a check is necessary in the first place (what if I want to search for all non-fork repositories, not just those owned by a particular user), but that's another story
- Assign `opts.Collaborate` inside the `switch modeQuery { ...`. If `repoOwner` turns out to be an organization, we can set `opts.Collaborate` to false then

I believe these changes would make the code a lot simpler.

Reply:
You are right, I was looking at it only from user persespective, but it can be more general.